### PR TITLE
ToDoList implementation

### PR DESCRIPTION
**Pull request by denwr2g**
ToDoList based on TableView

### CODE REVIEW

File: ToDoApp/AddItemViewController/AddItemViewController.swift [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+//
+//  Created by deniss.lobacs on 16/02/2022.
+//
+
+import UIKit
+import SnapKit
+
+class AddItemViewController: UIViewController {
+    
+    var addItemViewModel: AddItemViewModel?
```
Comment (eugenemokhan) [inline ToDoApp/AddItemViewController/AddItemViewController.swift]: let's use private, you have special functions to set this value

File: ToDoApp/AddItemViewController/AddItemViewController.swift [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        textField.textColor = .white
+        return textField
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureUI()
+    }
+    
+    func getItemName() -> String? {
```
Comment (eugenemokhan) [inline ToDoApp/AddItemViewController/AddItemViewController.swift]: why do you use this function? textFiles isn't private at your class

File: ToDoApp/AddItemViewController/AddItemViewController.swift (hunk 1 above)
Comment (eugenemokhan) [reply ToDoApp/AddItemViewController/AddItemViewController.swift]: let's use viewModel, haven't got another viewModels at this class

File: ToDoApp/AddItemViewController/AddItemViewController.swift [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    }
+    
+    @objc func goToListVC() {
+        addItemViewModel?.taskManager.addValue(value: textField.text ?? "")
+        addItemViewModel?.shouldOpenListVC()
+    }
+    
+}
+
+extension AddItemViewController {
```
Comment (eugenemokhan) [inline ToDoApp/AddItemViewController/AddItemViewController.swift]: for settings ui, better to use private extension

File: ToDoApp/AddItemViewController/AddItemViewController.swift [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    
+    func configureUI() {
+        view.backgroundColor = .purple
+        navigationItem.title = "Add item"
+        
+        view.addSubview(addItemButton)
+        addItemButton.snp.makeConstraints { make in
+            make.top.equalTo(view.safeAreaLayoutGuide).inset(50)
+            make.centerX.equalToSuperview()
```
Comment (eugenemokhan) [inline ToDoApp/AddItemViewController/AddItemViewController.swift]: where's bottom/height constraint for this view?

File: ToDoApp/AddItemViewController/AddItemViewController.swift [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        }
+        
+        view.addSubview(textField)
+        textField.snp.makeConstraints { make in
+            make.top.equalTo(addItemButton).offset(100)
+            make.centerX.equalToSuperview()
```
Comment (eugenemokhan) [inline ToDoApp/AddItemViewController/AddItemViewController.swift]: where's bottom/height constraint for this view?

File: ToDoApp/AddItemViewController/AddItemViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+//
+//  Created by deniss.lobacs on 16/02/2022.
+//
+
+import Foundation
+
+class AddItemViewModel {
+    
+    var onOpenListVC: ( () -> Void )?
+    var taskManager = TaskManager.shared
```
Comment (eugenemokhan) [inline ToDoApp/AddItemViewController/AddItemViewModel.swift]: let's use private

File: ToDoApp/AppDelegate.swift [hunk 7]
```diff
@@ -1,34 +1,52 @@
...
+        addItemViewModel.onOpenListVC = { [] in
+            navigationController.popToRootViewController(animated: true)
+        }
+        
+        
+        listViewController.configure(listViewModel: listViewModel)
+        addItemViewController.configure(addItemViewModel: addItemViewModel)
+        
+        window?.rootViewController = navigationController
+
```
Comment (eugenemokhan) [inline ToDoApp/AppDelegate.swift:51]: good job, but try to make 2 different functions for each viewController

File: ToDoApp/AppDelegate.swift [hunk 8]
```diff
@@ -1,34 +1,52 @@
...
-        return UISceneConfiguration(name: "Default Configuration", sessionRole: connectingSceneSession.role)
-    }
-
-    func application(_ application: UIApplication, didDiscardSceneSessions sceneSessions: Set<UISceneSession>) {
-        // Called when the user discards a scene session.
-        // If any sessions were discarded while the application was not running, this will be called shortly after application:didFinishLaunchingWithOptions.
-        // Use this method to release any resources that were specific to the discarded scenes, as they will not return.
+    
+    
+    func makeListViewController() -> UIViewController {
```
Comment (eugenemokhan) [inline ToDoApp/AppDelegate.swift]: delete it if you don't use it

File: ToDoApp/ListViewController/ListViewController.swift [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+//
+//  Created by deniss.lobacs on 15/02/2022.
+//
+
+import UIKit
+
+class ListViewController: UIViewController {
+    
+    var tableView = UITableView()
+    var listViewModel: ListViewModel?
```
Comment (eugenemokhan) [inline ToDoApp/ListViewController/ListViewController.swift]: let's use private

File: ToDoApp/ListViewController/ListViewController.swift [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configTable()
+    }
+        
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        tableView.reloadData()
```
Comment (eugenemokhan) [inline ToDoApp/ListViewController/ListViewController.swift]: why do you need it?

File: ToDoApp/ListViewController/ListViewController.swift [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+        self.tableView.delegate = self
+        self.tableView.dataSource = self
+        self.tableView.register(UINib(nibName: "TableCell", bundle: nil), forCellReuseIdentifier: "cell")
+        
+    }
+    
+    @objc private func goToAddListViewController() {
+        listViewModel?.shouldOpenAddVC()
+    }
+}
```
Comment (eugenemokhan) [inline ToDoApp/ListViewController/ListViewController.swift:72]: set this extension is above then delegates

File: ToDoApp/ListViewController/ListViewModel.swift [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+//
+//  Created by deniss.lobacs on 15/02/2022.
+//
+
+import Foundation
+
+class ListViewModel {
+    
+    var onOpenAddVC: ( () -> Void )?
+    var taskManager = TaskManager.shared
```
Comment (eugenemokhan) [inline ToDoApp/ListViewController/ListViewModel.swift]: let's use private

File: ToDoApp/Model/TaskManager.swift [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+//
+//  Created by deniss.lobacs on 16/02/2022.
+//
+
+import Foundation
+
+class TaskManager {
+    
+    static var shared = TaskManager()
+    var data: [String] = []
```
Comment (eugenemokhan) [inline ToDoApp/Model/TaskManager.swift]: mb let's use string?

File: ToDoApp/Model/TaskManager.swift [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+    
+    
+    private init(){}
+    
+    func addValue(value: String) {
```
Comment (eugenemokhan) [inline ToDoApp/Model/TaskManager.swift]: mb let's use string?

File: ToDoApp/Model/TaskManager.swift [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+        data.append(value)
+    }
+    
+    func getValue(index: Int) -> String {
```
Comment (eugenemokhan) [inline ToDoApp/Model/TaskManager.swift]: mb let's use string?

**Review by eugenemokhan — CHANGES_REQUESTED** [review]

File: ToDoApp/Model/TaskManager.swift (hunk 14 above)
Comment (denwr2g) [reply ToDoApp/Model/TaskManager.swift]: what should i use as string?

File: ToDoApp/Model/TaskManager.swift (hunk 15 above)
Comment (denwr2g) [reply ToDoApp/Model/TaskManager.swift]: what should i use as string?

File: ToDoApp/Model/TaskManager.swift (hunk 13 above)
Comment (denwr2g) [reply ToDoApp/Model/TaskManager.swift]: what should i use as string?

File: ToDoApp/ListViewController/ListViewController.swift (hunk 10 above)
Comment (denwr2g) [reply ToDoApp/ListViewController/ListViewController.swift]: for reloading tableview after appending inside new element

Comment (eugenemokhan) [reply ToDoApp/ListViewController/ListViewController.swift]: Are you sure, that this method will be called after appending new element? let's create a public function updateTableView() and call it when it necessary

File: ToDoApp/Model/TaskManager.swift (hunk 13 above)
Comment (eugenemokhan) [reply ToDoApp/Model/TaskManager.swift]: i mean optional value :)

File: ToDoApp/Model/TaskManager.swift (hunk 14 above)
Comment (eugenemokhan) [reply ToDoApp/Model/TaskManager.swift]: i mean optional value :)

File: ToDoApp/Model/TaskManager.swift (hunk 15 above)
Comment (eugenemokhan) [reply ToDoApp/Model/TaskManager.swift]: i mean optional value :)

File: ToDoApp/AddItemViewController/AddItemViewController.swift [hunk 16]
```diff
@@ -34,37 +45,51 @@ class AddItemViewController: UIViewController {
...
-        return textField.text
-    }
     
     func configure(addItemViewModel: AddItemViewModel) {
         self.addItemViewModel = addItemViewModel
     }
     
     @objc func goToListVC() {
-        addItemViewModel?.taskManager.addValue(value: textField.text ?? "")
+        addItemViewModel?.getTaskManager().addValue(value: textField.text ?? "")
```
Comment (eugenemokhan) [inline ToDoApp/AddItemViewController/AddItemViewController.swift]: let's create function at viewModel addValue(value: String) and call it from viewController, viewController shouldn't know about taskManager

File: ToDoApp/AddItemViewController/AddItemViewModel.swift [hunk 17]
```diff
@@ -10,10 +10,13 @@ import Foundation
...
     
     var onOpenListVC: ( () -> Void )?
-    var taskManager = TaskManager.shared
+    private var taskManager = TaskManager.shared
     
     func shouldOpenListVC() {
         self.onOpenListVC?()
     }
     
+    func getTaskManager() -> TaskManager {
```
Comment (eugenemokhan) [inline ToDoApp/AddItemViewController/AddItemViewModel.swift]: remove it pls

**Review by eugenemokhan — CHANGES_REQUESTED** [review]
Good job!

File: ToDoApp/ListViewController/ListViewModel.swift [hunk 18]
```diff
@@ -9,7 +9,7 @@ import Foundation
 
 class ListViewModel {
     
-    var onOpenAddVC: (() -> Void)?
+    var onOpenAddVC: ( () -> Void )?
```
Comment (eugenemokhan) [inline ToDoApp/ListViewController/ListViewModel.swift]: remove spaicngs pls, we don't use it in Swift

**Review by eugenemokhan — CHANGES_REQUESTED** [review]

**Pull request merged**